blocks/form/form.js

```diff
@@ -592,8 +592,12 @@ export default async function decorate(block) {
   let rules = true;
   let form;
   if (formDef) {
-    const { actionType, spreadsheetUrl } = formDef?.properties || {};
-    if (!formDef?.properties?.['fd:submit'] && actionType === 'spreadsheet' && spreadsheetUrl) {
+    const submitProps = formDef?.properties?.['fd:submit'];
+    const actionType = submitProps?.actionName || formDef?.properties?.actionType;
+    const spreadsheetUrl = submitProps?.spreadsheet?.spreadsheetUrl
+      || formDef?.properties?.spreadsheetUrl;
+
```

Comment:
@deepprakash345 We are giving priority to submit action saved via properties extension. Shouldn't we ideally hide the submission properties from property sheet going ahead and make extension the only preferred route?

My concern is an author might configure the url in wizard and try to edit it in properties sheet. In UI there should be only one place to configure this ideally.

Reply:
Hi @ravise5 , 
Whatever priority is there, that should reflect in the form definition generated and published. So that handling needs to be in the core components sling models.

This code is basically to read from the form definition. Previously the submitAction details were present as top level properties in the json. Now we have changed it to make sure the submit properties comes inside "fd:submit" namespace.

This code is just handling the newer one.
cc:- @vdua